src/Symfony/Component/HttpClient/RetryHttpClient.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpClient\Response\AsyncContext;
+use Symfony\Component\HttpClient\Response\AsyncResponse;
+use Symfony\Component\HttpClient\Retry\RetryStrategyInterface;
+use Symfony\Contracts\HttpClient\ChunkInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * Automatically retries the failing requests.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class RetryHttpClient implements HttpClientInterface
+{
+    use AsyncDecoratorTrait;
+
+    private $strategy;
+    private $logger;
+
+    private $retryCount = 0;
+
+    public function __construct(HttpClientInterface $client, RetryStrategyInterface $strategy, LoggerInterface $logger = null)
+    {
+        $this->client = $client;
+        $this->strategy = $strategy;
+        $this->logger = $logger ?: new NullLogger();
+    }
+
+    public function request(string $method, string $url, array $options = []): ResponseInterface
+    {
+        return new AsyncResponse($this->client, $method, $url, $options, function (ChunkInterface $chunk, AsyncContext $context) use ($method, $url, $options) {
+            $exception = null;
+            try {
+                // only retry first chunk
+                if ($chunk->isTimeout() || !$chunk->isFirst()) {
+                    yield $chunk;
+
+                    return;
+                }
+            } catch (TransportExceptionInterface $exception) {
+                // catch TransportExceptionInterface to send it to strategy.
```

Comment:
this will be reached also after the first chunk when the content stream breaks.
to prevent this, you can disable the passthru by calling `$context->passthru()` once the first chunk is yielded.

Reply:
Fixed